keras/src/layers/layer.py

```diff
@@ -1560,6 +1575,45 @@ def _open_name_scope(self):
             self._parent_path = current_path()
         return backend.name_scope(self.name, caller=self)
 
+    def rematerialized_call(self, layer_call, *args, **kwargs):
+        """Wrap the layer's call method to enable rematerialization dynamically.
+
+        Args:
+            layer_call: The original `call` method of a layer.
+
+        Returns:
+            callable: The wrapped method with rematerialization logic applied.
```

Comment:
Change documentation. This is not wrapping anymore. The return value is the result of calling `layer_call`.